src/runtime/virtcontainers/types/asset_test.go

```diff
@@ -79,23 +100,57 @@ func TestAssetNew(t *testing.T) {
 	_, err = tmpfile.Write(assetContent)
 	assert.Nil(err)
 
-	anno := map[string]string{
-		annotations.KernelPath: tmpfile.Name(),
-		annotations.KernelHash: assetContentHash,
+	type testData struct {
+		inputPathVar	string
+		inputHashVar	string
+		inputAssetType	AssetType
+		inputHash	string
+		expectError	bool
+		expectNilAsset	bool
 	}
-	a, err := NewAsset(anno, ImageAsset)
-	assert.Nil(err)
-	assert.Nil(a)
 
-	a, err = NewAsset(anno, KernelAsset)
-	assert.Nil(err)
-	assert.Equal(assetContentHash, a.computedHash)
-
-	anno = map[string]string{
-		annotations.KernelPath: tmpfile.Name(),
-		annotations.KernelHash: assetContentWrongHash,
+	data := []testData{
+		// Successful with correct hash
+		{annotations.KernelPath, annotations.KernelHash, KernelAsset, assetContentHash, false, false},
+		{annotations.ImagePath, annotations.ImageHash, ImageAsset, assetContentHash, false, false},
+		{annotations.InitrdPath, annotations.InitrdHash, InitrdAsset, assetContentHash, false, false},
+		{annotations.HypervisorPath, annotations.HypervisorHash, HypervisorAsset, assetContentHash, false, false},
+		{annotations.JailerPath, annotations.JailerHash, JailerAsset, assetContentHash, false, false},
+		{annotations.FirmwarePath, annotations.FirmwareHash, FirmwareAsset, assetContentHash, false, false},
+
+		// Failure with incorrect hash
+		{annotations.KernelPath, annotations.KernelHash, KernelAsset, assetContentWrongHash, true, false},
+		{annotations.ImagePath, annotations.ImageHash, ImageAsset, assetContentWrongHash, true, false},
+		{annotations.InitrdPath, annotations.InitrdHash, InitrdAsset, assetContentWrongHash, true, false},
+		{annotations.HypervisorPath, annotations.HypervisorHash, HypervisorAsset, assetContentWrongHash, true, false},
+		{annotations.JailerPath, annotations.JailerHash, JailerAsset, assetContentWrongHash, true, false},
+		{annotations.FirmwarePath, annotations.FirmwareHash, FirmwareAsset, assetContentWrongHash, true, false},
+
+		// Other failures
+		{annotations.KernelPath, annotations.KernelHash, ImageAsset, assetContentHash, false, true},
 	}
 
-	_, err = NewAsset(anno, KernelAsset)
-	assert.NotNil(err)
+	for _, d := range data {
+		anno := map[string]string{
+			d.inputPathVar: tmpfile.Name(),
+			d.inputHashVar: d.inputHash,
+		}
+
+		if d.expectNilAsset {
+			a, _ := NewAsset(anno, d.inputAssetType)
+			assert.Nil(err)
```

Comment:
- Nit: This isn't incorrect, but it might be clearer to make this `assert.NoError(err)`.
- Could you add a message to each `assert.*` call as shown in https://github.com/kata-containers/tests/blob/master/Unit-Test-Advice.md that includes `d` and the index number. That way, if one of these array-based tests fails, it will be very easy to identify *which* specific scenario is failing

Reply:
@jodh-intel thanks for the input, I applied these changes